File: C#/3/3_4.cs
int maxRetries = 3;
int attempt = 0;
string Connect = "Connected successfully.";
string RetryConnect = "Retrying connection...";
string FailedConnect = "Failed to connect after 3 attempts.";

while (attempt < maxRetries)
{
    if (TryConnect())
    {
        Console.WriteLine(Connect);
        break;
    }
    else
    {
        Console.WriteLine(RetryConnect);
        attempt++;
    }
}

if (attempt == maxRetries)
{
    Console.WriteLine(FailedConnect);
}

File: Python/3/3_4.py
max_retries = 3
attempt = 0
while attempt < max_retries:
    if try_connect():
        print("Connected successfully.")
        break
    else:
        print("Retrying connection...")
        attempt += 1

if attempt == max_retries:
    print("Failed to connect after 3 attempts.")
